src/prefect/agent/kubernetes/agent.py

```diff
@@ -802,6 +817,8 @@ def generate_deployment_yaml(
         agent_env[3]["value"] = image_pull_secrets or ""
         agent_env[4]["value"] = str(labels)
         agent_env[11]["value"] = backend
+        agent_env[13]["value"] = key
```

Comment:
why are these indexed at 13 and 14?

Reply:
I put them at the end to make as few changes as I could and there are some that are not set.